### ISSUE TITLE
Angular Limit from Pinjoint2D causes to rotate Node B erraticly.

### ISSUE THREAD

**Issue by LilacSyringa** [godotengine/godot#88481]
### Tested versions

Reproducable in: 
4.2.1.stable_mono
4.3-dev3_mono

### System information

Windows 10

### Issue description

If you create RigidBody2Ds, neither is a child of the other, and use Pinjoint2D to connect the two. The issue begins when you enable angular limit and set the upper and lower rotation limit, one of the nodes in the joint will start rotating around the joint uncontrollably.

In the video, the torque around the joint isn't building up immediately after it rotates around, it only starts to build up torque. This seems rather unpredictable.

When lower > upper || upper < lower, the node immediately builds up torque more and more unpredictably. Otherwise, other forces have to rotate the node around the joint before triggering the effect, with the two limits converging makes it more likely with less rotation required.

https://github.com/godotengine/godot/assets/138860645/3d411aa8-9a3f-477b-a438-3a56cb2b843a

### Steps to reproduce

Physics bodies and a Rigidybody connected with a pinjoint. 
Enable angular limit.
Set the upper and lower limit.

### Minimal reproduction project (MRP)

[TestProject.zip](https://github.com/godotengine/godot/files/14321747/TestProject.zip)

**Comment by SolidP** [comment]
Yup, been tearing my hair out with this one. In rapier2d it does work btw, but rapier is pretty unstable and crashes often + smoothness and bias on joints does nothing. Really hope they fix this, my ragdoll's head keeps spinning 360 degrees, looks so goofy.

**Comment by LilacSyringa** [comment]
> Yup, been tearing my hair out with this one. In rapier2d it does work btw, but rapier is pretty unstable and crashes often + smoothness and bias on joints does nothing. Really hope they fix this, my ragdoll's head keeps spinning 360 degrees, looks so goofy.

Yeah I'm aware. But damn, there's a bug there with raycasts. I thought of dealing with the broken pinjoint2D through integrated_physics but it kind of feels limiting.

**Comment by tomsterBG** [comment]
Still an issue in 4.4.1 stable on Linux without Mono (x11 Linux Mint, AMD CPU) on the 2D default physics engine with 0.1 angular damp and 0 gravity.

I suspect it's a negative sign somewhere and a wrapping error due to how it behaves. It seems to have an affinity for the forbidden area (beyond the limits) and here's how it behaves:
https://streamable.com/euo0dt
I am dragging with a static body at the mouse position and a spring of length 0 that connects the RigidBody2D and the StaticBody2D.

It might be good to try https://github.com/godotengine/godot/pull/104539 in my own fork to see how that'd behave.
